### DISCUSSION TITLE
excerpt only when text is longer than defined excerpt length

### DISCUSSION THREAD

**Discussion by 3muchfun** [GetPublii/Publii#1505] [Help] [answered]
Dear all,
I would like to add the condition to the following code, that the read more button shall not show up, when the text is shorter than the defined excerpt length. Any idea?

```
<div class="post-content">

    {{#if hasCustomExcerpt}}
        {{{excerpt}}}
    {{else}}
        {{{excerpt}}} {{/if}}
</div>

{{#if @config.custom.ReadMore}}
    <p>
        <a href="{{url}}" class="more-link">Continue reading</a>
    </p>
{{/if}}

**Comment by candidexmedia** [comment]
Did you solve this?

**Reply by 3muchfun** [reply, marked as answer]
Hi, @candidexmedia, 
yes, I did :-). I created a custom theme for Publli, and that's what I did:

1. In the post within the Publii App, I added the '_read more_' button where needed.
2. In my _postlist.hbs_ I added the following

```{{#if hasCustomExcerpt}}
        <p>{{{excerpt}}}</p>
        <p>
          <a href="{{url}}" class="more-link">Weiterlesen</a>
        </p>
        {{else}}
        {{{ text }}}
        {{/if}}
